packages/manager/src/features/Firewalls/components/FirewallSelect.tsx

```diff
@@ -47,9 +52,20 @@ interface Props<DisableClearable extends boolean>
 export const FirewallSelect = <DisableClearable extends boolean>(
   props: Props<DisableClearable>
 ) => {
-  const { errorText, hideDefaultChips, loading, value, ...rest } = props;
+  const {
+    errorText,
+    hideDefaultChips,
+    loading,
+    value,
+    options,
+    optionsFilter,
+    ...rest
+  } = props;
+
+  const { data: firewalls, error, isLoading } = useAllFirewallsQuery(!options);
 
-  const { data: firewalls, error, isLoading } = useAllFirewallsQuery();
+  const firewallOptions =
+    options || (optionsFilter ? firewalls?.filter(optionsFilter) : firewalls);
```

Comment:
Not really sure if it matters, but `??` might be better here

```suggestion
    options ?? (optionsFilter ? firewalls?.filter(optionsFilter) : firewalls);
```